src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2ConnectionTests.cs

```diff
@@ -468,9 +468,9 @@ public async Task StreamPool_StreamIsInvalidState_DontReturnedToPool()
             Assert.Equal(0, _connection.StreamPool.Count);
 
             var output = (Http2OutputProducer)stream.Output;
-            await output._dataWriteProcessingTask;
+            await output._dataWriteProcessingTask.DefaultTimeout();
 
-            await StopConnectionAsync(expectedLastStreamId: 1, ignoreNonGoAwayFrames: false);
+            await StopConnectionAsync(expectedLastStreamId: 1, ignoreNonGoAwayFrames: false).DefaultTimeout();
```

Comment:
StopConnectionAsync has a built-in timeout. I also generally don't bother putting timeouts in the app func since the client side of the test should timeout anyway.